### PR TITLE
Public form e2e

### PR DESCRIPTION
**Pull request by konzz**
fixes #

PR checklist:
- [ ] Update READ.me ?
- [ ] Update API documentation ?

QA checklist:
- [ ] Smoke test the functionality described in the issue
- [ ] Test for side effects
- [ ] UI responsiveness
- [ ] Cross browser testing
- [ ] Code review

### CODE REVIEW

File: app/api/auth/routes.js [hunk 1]
```diff
@@ -82,7 +82,8 @@ export default app => {
 
   app.get('/api/captcha', cors(corsOptions), async (_req, res) => {
     const captcha = svgCaptcha.create({ ignoreChars: '0OoiILluvUV' });
-    const storedCaptcha = await CaptchaModel.save({ text: captcha.text });
+    const text = process.env.DATABASE_NAME !== 'uwazi_e2e' ? captcha.text : '42hf';
```
Comment (daneryl) [inline app/api/auth/routes.js:85]: we sure we want to depend on the database_name ? i know the chances are small, but it does not feel right, are we using the same condition for the mail "mock" ?

File: e2e/suites/public-form.test.ts [hunk 2]
```diff
@@ -0,0 +1,81 @@
+import { adminLogin, logout } from '../helpers/login';
+import proxyMock from '../helpers/proxyMock';
+import insertFixtures from '../helpers/insertFixtures';
+import disableTransitions from '../helpers/disableTransitions';
+
+const selectors = {
+  pageContentsInput:
+    '.page-viewer.document-viewer > div > div.tab-content.tab-content-visible > textarea',
```
Comment (daneryl) [inline e2e/suites/public-form.test.ts:8]: i think we are abandoning this way of using selectors ? any specific reason to use the full path?, i believe there is only 1 or 2 textarea on the page ?

File: e2e/suites/public-form.test.ts [hunk 3]
```diff
@@ -0,0 +1,81 @@
...
+  useCustomLandingPage:
```
Comment (daneryl) [inline e2e/suites/public-form.test.ts:9]: this selector seems not used

File: e2e/suites/public-form.test.ts [hunk 4]
```diff
@@ -0,0 +1,81 @@
...
+    );
+
+    await expect(page).toClick('button', { text: 'Save' });
+  });
+
+  it('should visit the page and do a submit', async () => {
+    const [newTab] = await Promise.all([
+      new Promise(resolve => browser.once('targetcreated', target => resolve(target.page()))),
+      expect(page).toClick('a', { text: 'view page' }),
+    ]);
```
Comment (daneryl) [inline e2e/suites/public-form.test.ts:52]: this is difficult to understand, what does this do ? abstract into a function to give it meaning.

File: e2e/suites/public-form.test.ts [hunk 5]
```diff
@@ -0,0 +1,81 @@
...
+    );
+    await expect(newTab).toClick('span', { text: 'Bahamas' });
+    await expect(newTab).toFill('.captcha input', '42hf');
+    await expect(newTab).toClick('input[type="submit"]');
+    await expect(newTab).toClick('.alert.alert-success');
+  });
+
+  it('should check the newly created entity', async () => {
+    await page.bringToFront();
+    await page.waitFor(500); // index delay
```
Comment (daneryl) [inline e2e/suites/public-form.test.ts:67]: @fnocetti did you finally implement a function to force the elasticsearch refresh ?

**Review by daneryl — CHANGES_REQUESTED** [review]

File: app/api/auth/routes.js (hunk 1 above)
Comment (konzz) [reply app/api/auth/routes.js:85]: yes, its the same condition than in the mail mock, do not like it either

File: e2e/suites/public-form.test.ts (hunk 2 above)
Comment (konzz) [reply e2e/suites/public-form.test.ts:8]: Updated it to '.markdownEditor textarea'  thats the best I can do without modifying the Markdown component.
I think all components we write should accept className.

File: e2e/suites/public-form.test.ts (hunk 3 above)
Comment (konzz) [reply e2e/suites/public-form.test.ts:9]: removed

File: e2e/suites/public-form.test.ts (hunk 4 above)
Comment (konzz) [reply e2e/suites/public-form.test.ts]: extracted to a different method

File: e2e/suites/public-form.test.ts (hunk 5 above)
Comment (konzz) [reply e2e/suites/public-form.test.ts]: added @fnocetti  elastic refresh

File: app/api/auth/routes.js (hunk 1 above)
Comment (daneryl) [reply app/api/auth/routes.js:85]: lets go with it and create an tech debt to change both ?

**Review by daneryl — APPROVED** [review]

**Pull request merged**